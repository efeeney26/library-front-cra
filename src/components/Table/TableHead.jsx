import React from 'react'

const TableHead = (props) => {
  return (
    <tr>
      <th>Wow</th>
    </tr>
  )
}

export default TableHead
