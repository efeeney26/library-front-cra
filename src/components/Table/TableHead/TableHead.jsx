import React from 'react'
import PropTypes from 'prop-types'

import styles from './TableHead.module.css'

const TableHead = ({ data }) => (
  <tr>
    {data.map((item, i) => (
      <th
        // добавить nanoid
        key={i}
        className={styles.head}
      >
        {item}
      </th>
    ))}
  </tr>
)

TableHead.propTypes = {
  data: PropTypes.arrayOf(PropTypes.string)
}

TableHead.defaultProps = {
  data: []
}

export default TableHead
