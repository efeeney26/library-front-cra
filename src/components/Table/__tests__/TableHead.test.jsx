import React from 'react'
import TableHead from '../TableHead/TableHead'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

const props = {
  data: ['testHeader1', 'testHeader2']
}

test('TableHead snapshot', () => {
  const component = renderer.create(<TableHead {...props}/>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

describe('TableHead', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <TableHead {...props} />
    )
  })
  it('elements rendered: th', () => {
    expect(wrapper.find('th').length).toBe(2)
  })
})
