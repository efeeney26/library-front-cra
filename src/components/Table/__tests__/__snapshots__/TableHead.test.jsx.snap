// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead snapshot 1`] = `
<tr>
  <th
    className="head"
  >
    testHeader1
  </th>
  <th
    className="head"
  >
    testHeader2
  </th>
</tr>
`;
